tests/dualtor_mgmt/test_dualtor_linkmgrd_restart_mux_port_status.py

```diff
@@ -0,0 +1,107 @@
+from tests.common.dualtor.icmp_responder_control import start_icmp_responder                                # noqa: F401
+from tests.common.dualtor.mux_simulator_control import toggle_all_simulator_ports_to_upper_tor              # noqa: F401
+from tests.common.fixtures.ptfhost_utils import run_icmp_responder                                          # noqa: F401
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.utilities import wait_until
+from tests.conftest import rand_selected_dut                                                                # noqa: F401
+
+
+pytestmark = [
+    pytest.mark.topology("dualtor")
+]
+
+
+LOOP_TIMES_LEVEL_MAP = {
+    'debug': 1,
+    'basic': 10,
+    'confident': 50,
+    'thorough': 60,
+    'diagnose': 100
+}
+
+
+@pytest.fixture
+def loop_times(get_function_completeness_level):
+    normalized_level = get_function_completeness_level
+    if normalized_level is None:
+        normalized_level = 'debug'
+    return LOOP_TIMES_LEVEL_MAP[normalized_level]
+
+
+def check_mux_port_status_after_linkmgrd_restart(rand_selected_dut, ports, loop_times,              # noqa: F811
+                                                 status=None, health=None):
+    def _check_mux_port_status(duthost, ports, status, health):
+        show_mux_status_ret = show_muxcable_status(duthost)
+        logging.debug("show_mux_status_ret: {}".format(json.dumps(show_mux_status_ret, indent=4)))
+
+        for port in ports:
+            if port not in show_mux_status_ret:
+                return False
+
+            if health is None:
+                # Active-Active case
+                health = 'healthy' if status == 'active' else 'unhealthy'
+                if show_mux_status_ret[port]['status'] != status:
+                    logging.debug(f"Port {port} status-{show_mux_status_ret[port]['status']}, expected status-{status}")
+                    return False
+
+            if show_mux_status_ret[port]['health'] != health or show_mux_status_ret[port]['hwstatus'] != 'consistent':
+                logging.debug(f"Port {port} health-{show_mux_status_ret[port]['health']}, expected health-{health};"
+                              f"hwstatus-{show_mux_status_ret[port]['hwstatus']}, expected hwstatus-consistent")
+                return False
+        return True
+
+    for _ in range(loop_times):
+        rand_selected_dut.shell("docker exec mux supervisorctl restart linkmgrd")
+        pytest_assert(wait_until(30, 5, 0, lambda: "RUNNING" in rand_selected_dut.
+                                 command("docker exec mux supervisorctl status linkmgrd")
+                                 ["stdout"]), "linkmgrd is not running after restart")
+        pytest_assert(wait_until(120, 10, 0, _check_mux_port_status, rand_selected_dut, ports, status, health),
+                      "MUX port status is not correct after linkmgrd restart")
+
+
+@pytest.mark.parametrize("cable_type", ["active-active", "active-standby"])
```

Comment:
if you use `tests.common.dualtor.dual_tor_common.cable_type`, this is not needed, right?